### DISCUSSION TITLE
Change agent instructions with API

### DISCUSSION THREAD

**Discussion by sepa85** [langgenius/dify#5533] [Help]
I need to have the instructions of my agent dynamic. Is there a way to change the instructions via API?

Another option would be using the API based variables, but I tried without success. Is there a guide on how to use the API based variables? I only found tools related docs.

Thank you.

**Comment by crazywoola** [comment]
In workflow, you pass the prompt as a variable to LLM.

**Reply by sepa85** [reply]
For workflow I found that solution also. The problem for me are the instructions for the agent, that can accept variables, also from API, but i don't know how to get data from them and save into var.
